File: Source/JavaScript/Arc/queries/QueryHttpMethod.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

/**
 * Defines the HTTP method used to perform a query over HTTP.
 */
export enum QueryHttpMethod {
    /**
     * Use the GET method, carrying arguments in the URL query string (default).
     */
    Get = 'GET',

    /**
     * Use the QUERY method (RFC 10008), carrying arguments in a JSON request body.
     */
    Query = 'QUERY',

    /**
     * Prefer the QUERY method and automatically fall back to GET when the server or network path
     * does not support it. The outcome is remembered for the rest of the session so subsequent
     * queries go straight to the working transport.
     */
    Auto = 'AUTO',
}

File: Source/DotNET/Arc.Core/Queries/QueryHttpMethod.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Arc.Queries;

/// <summary>
/// Defines the HTTP method a generated client proxy uses to perform a query over HTTP.
/// </summary>
/// <remarks>
/// The member names match the TypeScript <c>QueryHttpMethod</c> enum so they can be carried through
/// proxy generation.
/// </remarks>
public enum QueryHttpMethod
{
    /// <summary>
    /// Use the GET method, carrying arguments in the URL query string.
    /// </summary>
    Get = 0,

    /// <summary>
    /// Use the QUERY method (RFC 10008), carrying arguments in a JSON request body.
    /// </summary>
    Query = 1,

    /// <summary>
    /// Prefer the QUERY method and fall back to GET when the server or network path does not support it.
    /// </summary>
    Auto = 2
}
